django/db/models/fields/related.py

```diff
@@ -1885,6 +1899,30 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required kwarg for %s in Django 2.0. "
+                "Set it to models.CASCADE if you want to maintain the current default behavior. "
+                "See https://docs.djangoproject.com/en/%s/ref/models/fields/"
+                "#django.db.models.ForeignKey.on_delete" % (
+                    self.__class__.__name__,
+                    get_docs_version(),
+                ),
+                RemovedInDjango20Warning, 2)
+            on_delete = CASCADE
+
+        if not callable(on_delete) and callable(to_field):
```

Comment:
Actually it should be `if on_delete and not callable(on_delete)` -- or just turn the `if` into an `elif`.